### ISSUE TITLE
`min` example for fusion/astdsl.nim is wrong

### ISSUE THREAD

**Issue by slonik-az** [nim-lang/fusion#43]
The `fusion/astdsl` has a usage example in https://nim-lang.github.io/fusion/src/fusion/astdsl.html
for the `min` macro that is wrong. 

### How to reproduce
replace the assertion in the example with
```Nim
assert min("d", "c", "b", "a") == "a"
```
Output
```
Error: unhandled exception: XXXX/min.nim(25, 8) `min("d", "c", "b", "a") == "a"`  [AssertionDefect]
```
The reason being that `min("d", "c", "b", "a")` returns "c" actually.

The min algorithm as implemented is wrong. It should not be a sequence of elif branches that stops the first time a smaller element is found. A loop over all the elements is needed to find the minimum.

**Comment by planetis-m** [comment]
Omg you are totally right!

**Comment by planetis-m** [comment]
The easiest way to implement min with a macro is using [nestList](https://github.com/planetis-m/looper/blob/23a04a4731d082a8ae490c727186f42509690517/looper.nim#L88) So maybe I should look for a better example.

**Comment by slonik-az** [comment]
Just sumbitted a PR #48  that fixes the `min` macro example code.

**Comment by narimiran** [comment]
Fixed in #48.
